src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -158,18 +159,19 @@ object StructuralVariant {
       val graph = MutableGraph[PairedMappedRead, WUnDiEdge]()
 
       for { (read, i) <- reads.zipWithIndex } {
-        val start = read.minPos
+        val (start, _, end, _) = read.startsAndStops
         var j = i + 1
         var done = false
         // Loop through subsequent reads until it's impossible to find another compatible one.
         while (j < reads.length && !done) {
           val nextRead = reads(j)
-          val nextStart = nextRead.minPos
+          val (nextStart, _, nextEnd, _) = nextRead.startsAndStops
           if (Math.abs(nextStart + nextRead.readLength - start) > maxNormalInsertSize) {
             done = true
           } else {
             if (areReadsCompatible(read, nextRead, maxNormalInsertSize)) {
-              graph += (read ~ nextRead) % 1
+              val weight = Math.abs((nextEnd - nextStart) - (end - start))
```

Comment:
what's going on here? difference of [distance from start of first read to start of second read] between the two pairs? "end" and "nextEnd" are not terribly clear; those are what I was calling "gap end"s right?